crypto/s2n_certificate.c

```diff
@@ -252,12 +356,22 @@ int s2n_cert_chain_and_key_free(struct s2n_cert_chain_and_key *cert_and_key)
         GUARD(s2n_free_object((uint8_t **)&cert_and_key->private_key, sizeof(s2n_cert_private_key)));
     }
 
-    if (cert_and_key->x509_cert) {
-        X509_free(cert_and_key->x509_cert);
+    if (cert_and_key->san_names) {
+        for (int i = 0; i < s2n_array_num_elements(cert_and_key->san_names); i++) {
+            struct s2n_blob *san_name = s2n_array_get(cert_and_key->san_names, i);
+            GUARD(s2n_free(san_name));
+        }
+        GUARD(s2n_array_free(cert_and_key->san_names));
+        cert_and_key->san_names = NULL;
     }
 
```

Comment:
Not quite because we have to call `s2n_free` on each blob element. The only way I can think of simplifying it is if we have a specialized `s2n_array` that is meant to only store `s2n_blobs`. Might be useful since I can imagine a lot of `s2n_array`s simply holding blobs.